kaggle_environments/core.py

```diff
@@ -573,29 +577,35 @@ def __process_specification(self, spec):
                 return ("type must be an integer or number", None)
             reward["type"] = [reward_type, "null"]
 
-        # Allow environments to extend the default configuration.
-        configuration = copy.deepcopy(
-            schemas["configuration"]["properties"])
-
-        for k, v in get(spec, dict, {}, ["configuration"]).items():
-            # Set a new default value.
-            if not isinstance(v, dict):
-                if not has(configuration, path=[k]):
+        # Allow environments to extend various parts of the specification.
+        def extend_specification(source, field_name):
```

Comment:
This section originally allowed us to define a generic configuration and an environment specific configuration and merge them together to form the final configuration. Now the same code is used to have a generic and environment specific version of any part of the specification -- in this case we've added a generic `observation.remainingOverageTime` field.